bindata/network/ovn-kubernetes-ipsec/ovn-ipsec.yaml

```diff
@@ -0,0 +1,142 @@
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: ovn-ipsec
+  namespace: openshift-ovn-kubernetes
+  annotations:
+    kubernetes.io/description: |
+      This DaemonSet launches the ovn ipsec networking components for all nodes.
+    release.openshift.io/version: "{{.ReleaseVersion}}"
+spec:
+  selector:
+    matchLabels:
+      app: ovn-ipsec
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: ovn-ipsec
+        component: network
+        type: infra
+        openshift.io/component: network
+        kubernetes.io/os: "linux"
+    spec:
+      serviceAccountName: ovn-kubernetes-node
+      hostNetwork: true
+      hostPID: true
+      priorityClassName: "system-node-critical"
+      # volumes in all containers:
+      # (container) -> (host)
+      # /etc/openvswitch -> /var/lib/openvswitch/etc - ovsdb system id
+      # /var/lib/openvswitch -> /var/lib/openvswitch/data - ovsdb data
+      # /run/openvswitch -> tmpfs - ovsdb sockets
+      # /env -> configmap env-overrides - debug overrides
+      containers:
+      # ovs-monitor-ipsec and libreswan daemons
+      - name: ovn-ipsec
+        image: "{{.OvnImage}}"
+        command:
+        - /bin/bash
+        - -c
+        - |
+          #!/bin/bash
+
+          function cleanup()
+          {
+            # Stop ovn-monitor-ipsec
+            /usr/share/openvswitch/scripts/ovs-ctl stop-ovs-ipsec
+            # Stop libreswan
+            /usr/libexec/ipsec/whack --shutdown
+            /sbin/ip xfrm policy flush
+            /sbin/ip xfrm state flush
+            /usr/sbin/ipsec --stopnflog
+            exit 0
+          }
+          trap cleanup SIGTERM
+
+          # Workaround for https://github.com/libreswan/libreswan/issues/373
+          ulimit -n 1024
+
+          /usr/libexec/ipsec/addconn --config /etc/ipsec.conf --checkconfig
+          # Check for kernel modules
+          /usr/libexec/ipsec/_stackmanager start
+          # Check for nss database status and migration
+          /usr/sbin/ipsec --checknss
+          # Check for nflog setup
+          /usr/sbin/ipsec --checknflog
+          # Start the actual IKE daemon
+          /usr/libexec/ipsec/pluto --leak-detective --config /etc/ipsec.conf --logfile /var/log/openvswitch/libreswan.log
+
+          # Required until latest upstream package is added to RHEL
+          curl https://github.com/lzhecheng/ovs/commit/869b06356e389079861962160e864df609d033e5.patch > ipsec.patch
```

Comment:
just FYI, we can't merge something like this.

If this won't be in RHEL in time, then we need to fix this in the dockerfile. And the patch needs to be in the repo; not downloaded.

Reply:
Yes, of course. This is currently building against a private ovn_k8s image. But this (and the environment variables a few lines below) will be fixed by this commit: https://github.com/openshift/ovn-kubernetes/pull/344

Reply:
I will remove this now as the openvswitch 2.13-ipsec package is available. Just waiting for this PR to merge: https://github.com/openshift/ovn-kubernetes/pull/344